### PR TITLE
publish docker image using circle ci sdk

### PR DESCRIPTION
**Pull request by Okhelifi**
## Issue

https://gravitee.atlassian.net/browse/APIM-2635

## Description

Here we finish the publish docker images workflow generation. To see what it will look like after the generation please refer to the dedicated unit test resources

### CODE REVIEW

File: .circleci/ci/src/jobs/job-publish-pr-env-urls.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+  public static create(dynamicConfig: Config): Job {
+    dynamicConfig.importOrb(orbs.keeper);
+    dynamicConfig.importOrb(orbs.github);
+
+    const notifyOnFailureCommand = NotifyOnFailureCommand.get(dynamicConfig);
+    dynamicConfig.addReusableCommand(notifyOnFailureCommand);
+
+    return new Job(PublishPrEnvUrlsJob.jobName, NodeLtsExecutor.create('small'), [
+      new commands.Checkout(),
```
Comment (phiz71) [inline .circleci/ci/src/jobs/job-publish-pr-env-urls.ts]: I'm wondering if we really need to checkout

**Review by jourdiw — APPROVED** [review]

File: .circleci/ci/src/pipelines/tests/pipeline-publish-docker-images.spec.ts [hunk 2]
```diff
@@ -15,6 +15,7 @@ describe('Publish docker images tests', () => {
       buildNum: '1234',
       buildId: '1234',
       graviteeioVersion: '4.2.0',
+      apimVersionPath: './src/../../../pom.xml',
```
Comment (phiz71) [inline .circleci/ci/src/pipelines/tests/pipeline-publish-docker-images.spec.ts]: Tests will be wrong after each release, won't they ?

Comment (phiz71) [reply .circleci/ci/src/pipelines/tests/pipeline-publish-docker-images.spec.ts]: in this line:
`command: "sed -i 's/\"version\": \".*\"/\"version\": \"4.2.0-SNAPSHOT\"/' << parameters.apim-ui-project >>/build.json"`

File: .circleci/ci/src/workflows/workflow-publish-docker-images.ts [hunk 3]
```diff
@@ -11,17 +12,46 @@ export class PublishDockerImagesWorkflow {
...
         context: config.jobContext,
         requires: ['Build backend'],
         name: 'Build and push rest api and gateway images',
       }),
+      new workflow.WorkflowJob(webuiBuildJob, {
+        context: config.jobContext,
+        requires: ['Setup'],
+        name: 'Build APIM Console and publish image',
+        'apim-ui-project': 'gravitee-apim-console-webui',
+        'docker-image-name': 'apim-management-ui',
```
Comment (phiz71) [inline .circleci/ci/src/workflows/workflow-publish-docker-images.ts]: you can use the config object here

Comment (Okhelifi) [reply .circleci/ci/src/workflows/workflow-publish-docker-images.ts]: done

File: .circleci/ci/src/pipelines/tests/pipeline-publish-docker-images.spec.ts (hunk 2 above)
Comment (Okhelifi) [reply .circleci/ci/src/pipelines/tests/pipeline-publish-docker-images.spec.ts]: done

File: .circleci/ci/src/jobs/job-publish-pr-env-urls.ts (hunk 1 above)
Comment (Okhelifi) [reply .circleci/ci/src/jobs/job-publish-pr-env-urls.ts]: done

**Review by phiz71 — APPROVED** [review]

**Pull request merged**